### PR TITLE
Add validation for VK_EXT_pipeline_protected_access

### PR DESCRIPTION
**Pull request by Tony-LunarG**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -3529,6 +3530,39 @@ bool CoreChecks::ValidatePipeline(std::vector<std::shared_ptr<PIPELINE_STATE>> c
                                          "VkPipelineShaderStageModuleIdentifierCreateInfoEXT and identifierSize not equal to 0",
                                          pipe_index, string_VkGraphicsPipelineLibraryFlagsEXT(lib->graphics_lib_type).c_str());
                     }
+
+                    std::vector<std::pair<VkPipelineCreateFlagBits, std::pair<std::string, std::string>>> test_infos = {
```
Comment (jeremyg-lunarg) [inline layers/core_validation.cpp:3534]: this should be static const, if possible and maybe an array instead of a vector.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM once test_infos is consty

File: layers/core_validation.cpp (hunk 1 above)
Comment (juan-lunarg) [reply layers/core_validation.cpp:3534]: constexpr std::array?

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -3529,6 +3530,39 @@ bool CoreChecks::ValidatePipeline(std::vector<std::shared_ptr<PIPELINE_STATE>> c
                                          "VkPipelineShaderStageModuleIdentifierCreateInfoEXT and identifierSize not equal to 0",
                                          pipe_index, string_VkGraphicsPipelineLibraryFlagsEXT(lib->graphics_lib_type).c_str());
                     }
+
+                    static const std::array<std::pair<VkPipelineCreateFlagBits, std::pair<std::string, std::string>>, 2> check_infos =
+                        {std::make_pair(VK_PIPELINE_CREATE_NO_PROTECTED_ACCESS_BIT_EXT,
+                                        std::make_pair("VUID-VkPipelineLibraryCreateInfoKHR-pipeline-07404",
+                                                       "VUID-VkPipelineLibraryCreateInfoKHR-pipeline-07405")),
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:3537]: Might be worth creating a struct just above to make this a bit clearer what each field is vs nested pairs, but I'm good either way here.

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

File: layers/core_validation.cpp (hunk 1 above)
Comment (Tony-LunarG) [reply layers/core_validation.cpp:3534]: Fixed

File: layers/core_validation.cpp (hunk 2 above)
Comment (Tony-LunarG) [reply layers/core_validation.cpp]: Fixed

**Pull request merged**